crates/connector_proxy/src/main.rs

```diff
@@ -114,54 +128,99 @@ async fn signal_handler() {
+        }
+        ProxyCommand::DelayedExecute(ba) => delayed_execute(ba.config_file_path).await,
     }
 }
 
-async fn proxy_flow_capture(c: ProxyFlowCapture, image_config: ImageConfig) -> Result<(), Error> {
-    let mut converter_pair = match image_config
-        .get_connector_protocol::<CaptureConnectorProtocol>(CaptureConnectorProtocol::Airbyte)
-    {
-        CaptureConnectorProtocol::FlowCapture => DefaultFlowCaptureInterceptor::get_converters(),
-        CaptureConnectorProtocol::Airbyte => AirbyteCaptureInterceptor::get_converters(),
+async fn proxy_flow_capture(
+    c: ProxyFlowCapture,
+    image_inspect_json_path: Option<String>,
+) -> Result<(), Error> {
+    let image_config = ImageConfig::parse_from_json_file(image_inspect_json_path)?;
+
+    let mut interceptor: Box<dyn Interceptor<FlowCaptureOperation>> = match image_config
+        .get_connector_protocol::<CaptureConnectorProtocol>(
+        CaptureConnectorProtocol::Airbyte,
+    ) {
+        CaptureConnectorProtocol::FlowCapture => {
+            Box::new(DefaultFlowCaptureInterceptor { _type: PhantomData })
+        }
+        CaptureConnectorProtocol::Airbyte => Box::new(AirbyteCaptureInterceptor::new()),
     };
 
-    converter_pair = compose(
-        converter_pair,
-        NetworkProxyCaptureInterceptor::get_converters(),
-    );
+    interceptor = compose(interceptor, Box::new(NetworkProxyCaptureInterceptor {}));
 
     run_connector::<FlowCaptureOperation>(
         c.operation,
         image_config.get_entrypoint(vec![DEFAULT_CONNECTOR_ENTRYPOINT.to_string()]),
-        converter_pair,
+        interceptor,
     )
     .await
 }
 
 async fn proxy_flow_materialize(
     m: ProxyFlowMaterialize,
-    image_config: ImageConfig,
+    image_inspect_json_path: Option<String>,
 ) -> Result<(), Error> {
+    // Respond to OS sigterm signal.
+    tokio::task::spawn(async move { sigterm_handler().await });
+
+    let image_config = ImageConfig::parse_from_json_file(image_inspect_json_path)?;
+
     // There is only one type of connector protocol for flow materialize.
-    let mut converter_pair = DefaultFlowMaterializeInterceptor::get_converters();
-    converter_pair = compose(
-        converter_pair,
-        NetworkProxyMaterializeInterceptor::get_converters(),
-    );
+    let mut interceptor: Box<dyn Interceptor<FlowMaterializeOperation>> =
+        Box::new(DefaultFlowMaterializeInterceptor { _type: PhantomData });
+    interceptor = compose(interceptor, Box::new(NetworkProxyMaterializeInterceptor {}));
 
     run_connector::<FlowMaterializeOperation>(
         m.operation,
         image_config.get_entrypoint(vec![DEFAULT_CONNECTOR_ENTRYPOINT.to_string()]),
-        converter_pair,
+        interceptor,
     )
     .await
 }
+
+async fn delayed_execute(command_config_path: String) -> Result<(), Error> {
+    // Sleep for some time to allow parent process to stop the current process.
+    std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
This was a bit hacky. 
The approach I've tried is - let the `delayed_execute` process pause itself, and wait for `sigcont` signal from its parent (connector proxy) process to continue. The issue here was that the - `sigcont` from the parent might came even before the `delayed_execute` stopped and prepared to listen to the `sigcont` signal. If `delayed_execute` stopped after missing the `sigcont` signal, it will stack forever.
The main issue is that - separate the stop/resume actions in two processes is not good, b/c it is difficult to coordinate them.

So in this implementation, both the `stop` and `start` signals are coming from the parent process, so that they could be ordered. And added the sleep logic in the delayed process to given enough time for the parent process to stop it. Alternatives I was thinking -
- Using a locker file. The parent creates a locker file before starting the child process. And the child process will block and ping the locker file until it is deleted.
- Building a binary using other languages, and inject it to the docker image together with `connector-proxy`.  (Or calling to the other languages from Rust). The process API in Rust does not allow us to do much. Hopefully the other language could.